### ISSUE TITLE
If product has negative stock, shipment and credit can't be created in Magento 2.4.3

### ISSUE THREAD

**Issue by mdesplenter** [magento/magento2#34250]
If a product has negative stock, shipment and credit can't be created in Magento 2.4.3

### Preconditions (*)
1. Magento 2.4.3

### Steps to reproduce (*)
1. Create an order
2. Set stock of product to negative value (fe. -3)
3. Create a shipment
4. or after the order is finished, create a credit, while putting back the product into stock.

### Expected result (*)
1. Shipment/credit being created

### Actual result (*)
1. Shows alert that the product is not in stock.

### Workaround
1. Before creating the shipment or credit, set a positive value in the product stock.
---
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._

**Comment by mdesplenter** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Delta** [comment]
@magento give me 2.4-develop instance

**Comment by mdesplenter** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Delta** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Delta** [comment]
Hi @mdesplenter,

We tried to reproduce the issue as per provided steps but it is not replicated in Magento 2.4 Develop.
Hence, request to have a look into it and confirm us.
Added the label 'Needs Update.
Kindly find the attached screenshots for reference.
![quantity_update_Negative](https://user-images.githubusercontent.com/51681379/136151533-de125ddc-b5b8-4dfe-a461-28b6a177defe.JPG)
![shipment_created](https://user-images.githubusercontent.com/51681379/136151567-83bf8a0f-2cd5-4a40-aa8d-56047c208d00.JPG)
'
